diff -urN amavisd-snapshot-20020300.org/configure.in amavisd-snapshot-20020300/configure.in
--- amavisd-snapshot-20020300.org/configure.in	Fri Jun  7 01:35:35 2002
+++ amavisd-snapshot-20020300/configure.in	Fri Jun  7 01:54:16 2002
@@ -160,16 +160,6 @@
 qmaildir=`dirname $qmailinject`
 AC_SUBST(qmaildir)
 
-if test "x$qmailinject" != "x/bin/false"; then
-  AC_ARG_ENABLE(qmail,
-    [  --enable-qmail                use qmail as MTA],
-    if test "x$enableval" != "xno" ; then
-      MTA=qmail
-    fi,
-    MTA=qmail
-    config_type=qmail-queue)
-fi
-
 #
 if test "x$MTA" = "xqmail" ; then
   AC_MSG_ERROR([qmail is not supported by amavisd])
@@ -243,28 +233,12 @@
     MTA=postfix
   fi)
 
-dnl Need to find out whether sendmail_wrapper is really postfix' sendmail
-if test "x$MTA" = "xsendmail" ; then
-  strings ${sendmail_wrapper} | grep -i postfix >/dev/null 2>&1
-  if test $? -eq 0 ; then
-    MTA=postfix
-  fi
-fi
-
 AC_ARG_ENABLE(exim,
   [  --enable-exim                 use exim as MTA],
   if test "x$enableval" = "xyes" ; then
     MTA=exim
   fi)
 
-dnl Need to find out whether sendmail_wrapper is really exim's sendmail
-if test "x$MTA" = "xsendmail" ; then
-  strings ${sendmail_wrapper} | grep -i exim >/dev/null 2>&1
-  if test $? -eq 0 ; then
-    MTA=exim
-  fi
-fi
-
 dnl only sendmail
 dnl relay/no relay must be known before looking
 dnl for original sendmail.cf file
