diff -Nru amavisd-snapshot-20010714.orig/configure.in amavisd-snapshot-20010714/configure.in
--- amavisd-snapshot-20010714.orig/configure.in	Sat Jul 14 18:23:35 2001
+++ amavisd-snapshot-20010714/configure.in	Sat Apr 20 15:05:49 2002
@@ -217,12 +217,12 @@
   fi)
 
 dnl Need to find out whether sendmail_wrapper is really postfix' sendmail
-if test "x$MTA" = "xsendmail" ; then
-  strings ${sendmail_wrapper} | grep -i postfix >/dev/null 2>&1
-  if test $? -eq 0 ; then
-    MTA=postfix
-  fi
-fi
+dnl if test "x$MTA" = "xsendmail" ; then
+dnl   strings ${sendmail_wrapper} | grep -i postfix >/dev/null 2>&1
+dnl   if test $? -eq 0 ; then
+dnl     MTA=postfix
+dnl   fi
+dnl fi
 
 AC_ARG_ENABLE(exim,
   [  --enable-exim                 use exim as MTA],
@@ -231,12 +231,12 @@
   fi)
 
 dnl Need to find out whether sendmail_wrapper is really exim's sendmail
-if test "x$MTA" = "xsendmail" ; then
-  strings ${sendmail_wrapper} | grep -i exim >/dev/null 2>&1
-  if test $? -eq 0 ; then
-    MTA=exim
-  fi
-fi
+dnl if test "x$MTA" = "xsendmail" ; then
+dnl   strings ${sendmail_wrapper} | grep -i exim >/dev/null 2>&1
+dnl   if test $? -eq 0 ; then
+dnl     MTA=exim
+dnl   fi
+dnl fi
 
 dnl
 dnl only postfix and sendmail for now
