diff -Nru amavisd-snapshot-20010714.orig/configure.in amavisd-snapshot-20010714/configure.in
--- amavisd-snapshot-20010714.orig/configure.in	Sat Jul 14 18:23:35 2001
+++ amavisd-snapshot-20010714/configure.in	Sat Apr 20 16:47:21 2002
@@ -173,28 +173,7 @@
 
 if test "x$enable_milter" = xyes ; then
 
-  AC_CHECK_FUNCS(inet_aton,,AC_CHECK_LIB(resolv,inet_aton))
-
-  ACX_PTHREAD([
-    LIBS="$PTHREAD_LIBS $LIBS"
-    CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
-    CC="$PTHREAD_CC"
-
-    AC_CHECK_HEADERS(sm/gen.h,
-      AC_EGREP_HEADER([typedef.*bool],sm/gen.h,
-        AC_DEFINE(HAVE_SM_GEN_BOOL_TYPE, 1,
-                  [ Define if <sm/gen.h> defines the bool type. ])))
-
-    # sendmail 8.11: need -lsmutil
-    # sendmail 8.12: need -lsm
-    AC_CHECK_LIB(smutil,errstring,,AC_CHECK_LIB(sm,sm_errstring))
-
-    AC_CHECK_HEADERS(libmilter/mfapi.h,
-      AC_CHECK_LIB(milter,smfi_main,,enable_milter=no),
-      enable_milter=no)
-    ],
-    [AC_MSG_WARN([No usable pthreads library found - disabling milter])
-     enable_milter=no])
+     enable_milter=no
 
 fi
 
